src/pokemon.c

```diff
@@ -7243,40 +7244,43 @@ bool8 TryIncrementMonLevel(struct Pokemon *mon)
     }
 }
 
-u32 CanMonLearnTMHM(struct Pokemon *mon, u8 tm)
+u32 CanMonLearnTaughtMove(struct Pokemon *mon, u16 move)
 {
     u16 species = GetMonData(mon, MON_DATA_SPECIES2, 0);
```

Comment:
Wouldn't it make sense to make it so that CanMonLearnTaughtMove calls CanSpeciesLearnTaughtMove instead of duplicating code two times?

Reply:
Definitely. Left it as-is to avoid additional opinionated changes, but I'm more than willing to make them.

I was considering just removing CanMonLearnTaughtMove entirely - the only thing it does differently is grabs the species from the mon via `GetMonData(mon, MON_DATA_SPECIES2, 0);` and that code can be added to the only 2 sections that use the function (CanMonLearnMove in party_menu.c and BuildEggMoveset in daycare.c). Not really a big difference either way - but imo it's a better practice to do data conversion like that in-context _before_ passing the value into a function instead of having multiple functions that do conversion in different ways. GetMonData is already widely used around the codebase to retrieve MON_DATA_SPECIES2 (93 results for uses with that enum member specifically) so I figure it'd be consistent.

Thoughts?

Reply:
Sure, go ahead

Reply:
Changed in https://github.com/rh-hideout/pokeemerald-expansion/pull/2233/commits/b8a3cc284650eebe2357986b07ed14fe1f94a5c2.